src/main/src/main/java/org/geoserver/catalog/impl/AttributeTypeInfoImpl.java

```diff
@@ -141,7 +141,7 @@ public String getSource() {
                 return name;
             } catch (CQLException e) {
                 // quoting to avoid reserved keyword issues
-                return "\"" + name + "\"";
+                return "'" + name.replace("'", "''") + "'";
```

Comment:
To create a ``PropertyName`` expression that can evaluate the source attribute contents, rather than a string.
If the "weird attribute column names" do not work, not even when double-quoted, then they might be using chars that go outside of the CQL grammar (just making a guess). 

Did you try the fix in this PR? Are WMS/WFS services actually working with it, and returning the attribute values, or are they just giving you back always the same string value (which happens to be the attribute name)?

Reply:
OK, I still have to try, if my problem with uploading a GeoPackage containing wired column names is solved by this (will try tomorrow).

I did not intend to run any CQL on my attributes. The braces came in randomly.

But the new feature, to rename column names (GEOS-10356),  sounds promising.
So I may use simple technical names for the columns instead and rename them into a complex description to be shown up on the legend later on.

Reply:
OK, got some time for an explanation:

https://osgeo-org.atlassian.net/browse/GEOS-10577?focusedCommentId=83736